diff --git a/mew.c b/mew.c
index 1452bd3..dfb8bec 100644
--- a/mew.c
+++ b/mew.c
@@ -499,7 +499,6 @@ keyboard_keypress(enum wl_keyboard_key_state state, xkb_keysym_t sym)
 		case XKB_KEY_f: sym = XKB_KEY_BackSpace; break;
 		case XKB_KEY_g: sym = XKB_KEY_Escape; break;
 		case XKB_KEY_h: sym = XKB_KEY_BackSpace; break;
-		case XKB_KEY_i: sym = XKB_KEY_Tab; break;
 		case XKB_KEY_j: /* fallthrough */
 		case XKB_KEY_J: /* fallthrough */
 		case XKB_KEY_m: /* fallthrough */
@@ -507,6 +506,14 @@ keyboard_keypress(enum wl_keyboard_key_state state, xkb_keysym_t sym)
 		case XKB_KEY_n: sym = XKB_KEY_Down; break;
 		case XKB_KEY_p: sym = XKB_KEY_Up; break;
 
+		case XKB_KEY_i: /* copy selected item */
+			if (!sel)
+				return;
+			cursor = strnlen(sel->text, sizeof text - 1);
+			memcpy(text, sel->text, cursor);
+			text[cursor] = '\0';
+			match();
+			goto draw;
 		case XKB_KEY_k: /* delete right */
 			text[cursor] = '\0';
 			match();
@@ -551,10 +558,10 @@ keyboard_keypress(enum wl_keyboard_key_state state, xkb_keysym_t sym)
 			goto draw;
 		case XKB_KEY_g: sym = XKB_KEY_Home;  break;
 		case XKB_KEY_G: sym = XKB_KEY_End;   break;
-		case XKB_KEY_h: sym = XKB_KEY_Up;    break;
-		case XKB_KEY_j: sym = XKB_KEY_Next;  break;
-		case XKB_KEY_k: sym = XKB_KEY_Prior; break;
-		case XKB_KEY_l: sym = XKB_KEY_Down;  break;
+		case XKB_KEY_h: sym = XKB_KEY_Prior; break;
+		case XKB_KEY_j: sym = XKB_KEY_Down;  break;
+		case XKB_KEY_k: sym = XKB_KEY_Up;    break;
+		case XKB_KEY_l: sym = XKB_KEY_Next;  break;
 		default:
 			return;
 		}
@@ -612,6 +619,7 @@ keyboard_keypress(enum wl_keyboard_key_state state, xkb_keysym_t sym)
 		/* fallthrough */
 	case XKB_KEY_Up:
 	case XKB_KEY_KP_Up:
+	case XKB_KEY_ISO_Left_Tab:
 		if (sel && sel->left && (sel = sel->left)->right == curr) {
 			curr = prev;
 			calcoffsets();
@@ -652,19 +660,12 @@ keyboard_keypress(enum wl_keyboard_key_state state, xkb_keysym_t sym)
 		/* fallthrough */
 	case XKB_KEY_Down:
 	case XKB_KEY_KP_Down:
+	case XKB_KEY_Tab:
 		if (sel && sel->right && (sel = sel->right) == next) {
 			curr = next;
 			calcoffsets();
 		}
 		break;
-	case XKB_KEY_Tab:
-		if (!sel)
-			return;
-		cursor = strnlen(sel->text, sizeof text - 1);
-		memcpy(text, sel->text, cursor);
-		text[cursor] = '\0';
-		match();
-		break;
 	default:
 		if (xkb_keysym_to_utf8(sym, buf, 8))
 			insert(buf, strnlen(buf, 8));
